server/apparmor/apparmor_supported.go

```diff
@@ -47,13 +36,11 @@ type profileData struct {
 	Version int
 }
 
-// InstallDefaultAppArmorProfile installs default apparmor profile.
-func InstallDefaultAppArmorProfile() {
-	if err := InstallDefault(defaultApparmorProfile); err != nil {
-		// Allow daemon to run if loading failed, but are active
-		// (possibly through another run, manually, or via system startup)
-		if !IsLoaded(defaultApparmorProfile) {
-			logrus.Errorf("AppArmor enabled on system but the %s profile could not be loaded.", defaultApparmorProfile)
+// LoadDefaultAppArmorProfile loads default apparmor profile, if it is not loaded.
+func LoadDefaultAppArmorProfile() {
+	if !IsLoaded(DefaultApparmorProfile) {
```

Comment:
Several things that are wrong with this code:

1. This doesn't return an error. My changes in docker/docker#29130 fix this.
2. `IsLoaded` still doesn't differentiate between "an error occurred" and "no error occurred but the profile wasn't loaded". Also fixed in docker/docker#29130.
3. `IsLoaded` has a timeout -- which doesn't make sense because it's a kernel interface. There's nothing to DoS us.
4. The configured profile name isn't actually passed to this function, which means that the configuration is actually ignored.